.github/actions/setup_ubuntu_build/action.yml

```diff
@@ -0,0 +1,47 @@
+name: setup-ubuntu-build
+description: Setup ubuntu ready for building. Includes installs, ninja, build cache setup and loading llvm cache
+
+inputs:
+  build_type:
+    description: 'build type (Release, RelAssert)'
+    default: RelAssert
```

Comment:
It's a shame that it's `RelAssert` and not `ReleaseAssert` for consistency. Are we at all in control of that?

Reply:
Also this should maybe say `llvm_build_type` to be clear.

Reply:
I had issues with patterns on the caches

Reply:
I believe we could possible make it use ReleaseAssert if we changed the ordering in the cache name, but I think that should be left for  a later commit as it would require rebuilding the llvm caches.

Reply:
I'll look into this at a later point.

Reply:
Aye fine